frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1874,7 +1874,8 @@ function ReaderHighlight:onHighlightPress()
                 -- no existing highlight at current indicator position: start hold
                 self._start_indicator_highlight = true
                 self:onHold(nil, self:_createHighlightGesture("hold"))
-                if self.selected_text and self.selected_text.sboxes and #self.selected_text.sboxes then
+                -- With crengine, selected_text.sboxes does return good coordinates.
+                if not self.ui.document.info.has_pages and self.selected_text and self.selected_text.sboxes and #self.selected_text.sboxes > 0 then
```

Comment:
You will meet a lot of `if / if not self.ui.document.info.has_pages` in the code , but the prefered rule for new code is to use now the cleaner alternatives:
```
if self.ui.document.info.has_pages      => if self.ui.paging
if not self.ui.document.info.has_pages  => if self.ui.rolling
```

Reply:
Well, dynamic languages seems a lot of magic there.